osgp/platform/osgp-core/src/main/java/org/opensmartgridplatform/core/application/services/EventNotificationHelperService.java

```diff
@@ -63,6 +63,14 @@ public Ssld findSsld(final Long id) throws UnknownEntityException {
         return ssld.get();
     }
 
+    public Ssld findSsld(final String deviceIdentification) throws UnknownEntityException {
+        final Ssld ssld = this.ssldRepository.findByDeviceIdentification(deviceIdentification);
```

Comment:
Not in scope for this story, but it would be nice if the `findByDeviceIdentification` call is modified to return an `Optional<Ssld>`...

Reply:
Agreed to look into this some other time.